Issue #3246384 by vnech: Move unapplied code from Album config override

nechai opened the pull request:
## Problem
In the config overrides exists code that never applied:
- `modules/social_features/social_post/modules/social_post_album/src/SocialPostPhotoAlbumConfigOverride.php`

Also,  during updating OS to 10.3 a run time error is rising (see screenshot).

## Solution
- Move unapplied code from config overrides to appropriate hooks etc.

## Issue tracker
- https://www.drupal.org/project/social/issues/3246384

## How to test
- [x] Login as SM
- [x] Create "photo" post
- [x] Go to canonical post page
- [ ] Make sure the images in post using `social_x_large` image style

## Screenshots
![image](https://user-images.githubusercontent.com/19921926/139246368-c6938cd6-00f9-4e2f-af4b-b5edf1544bd9.png)

## Release notes
*Move unapplied code from config overrides to appropriate hooks etc.*

## Change Record
N/A

## Translations
N/A

ribel reviewed (approved):
This PR reverts fatal errors after the update to 10.2.6 or 10.3.0 which was introduced in #2493 and solves it in a different way.

Ressinel commented:
Hi, @nechai Thanks for your fixes!!!

Could you add the HTT list?

Do we need to wait for fixes for the two first config overrides before review since I see fixes for the last one only?
 - modules/social_features/social_event/modules/social_event_invite/src/SocialEventInviteConfigOverride.php
- modules/social_features/social_group/modules/social_group_invite/src/SocialGroupInviteConfigOverride.php

nechai commented:
@Ressinel I updated the PR description, please check. 
For [these two configs overrides services](https://github.com/goalgorilla/open_social/pull/2569#issuecomment-959923576) would be great to add fixes as each one uses `requestStack` service and I'm not sure that these changes are applying as configs are cached 🤔... But as they don't cause any issues I think fixes for them can be postponed and added in another PR.

Ressinel reviewed (approved):
Looks good to me.

ribel commented:
🍒 picked to 10.3.x and 10.2.x
